raiden/utils/upgrades.py

```diff
@@ -133,94 +150,91 @@ class UpgradeManager:
         match = VERSION_RE.match(base_name)
         assert match, f'Database name "{base_name}" does not match our format'
 
-        self._current_version = match.group(1)
         self._current_db_filename = Path(db_filename)
+        self._current_version = get_file_version(db_filename)
         self._kwargs = kwargs
 
     def run(self):
-        """
-        The `_current_db_filename` is going to hold the filename of the database
-        with the new version. However, the previous version's data
-        is going to exist in a file whose name contains the old version.
-        Therefore, running the migration means that we have to copy
-        all data to the current version's database and execute the migration
-        functions.
-        """
+        # First clear up any partially upgraded databases.
+        #
+        # A database will be partially upgraded if the process receives a
+        # SIGKILL/SIGINT while executing migrations. NOTE: It's very probable
+        # the content of the database remains consistent, because the upgrades
+        # are executed inside a migration, however making a second copy of the
+        # database does no harm.
         paths = glob(f'{self._current_db_filename.parent}/v*_log.db')
-        older_file = older_db_file(paths)
-
-        if older_file is None or older_file == str(self._current_db_filename):
-            return
+        valid_db_names = filter_db_names(paths)
+        delete_dbs_with_failed_migrations(valid_db_names)
 
-        old_db_filename = Path(older_file)
-
-        with get_file_lock(old_db_filename), get_file_lock(self._current_db_filename):
-            if self._current_db_filename.exists():
-                db_version = get_db_version(self._current_db_filename)
+        # At this point we know every file version and db version match
+        # (assuming there are no concurrent runs).
+        paths = glob(f'{self._current_db_filename.parent}/v*_log.db')
+        valid_db_names = filter_db_names(paths)
+        latest_db_path = latest_db_file(valid_db_names)
 
-                # The current version matches our target version, nothing to do.
-                if db_version == RAIDEN_DB_VERSION:
-                    return
+        # First run, there is no database file available
+        if latest_db_path is None:
+            return
 
-                if db_version > RAIDEN_DB_VERSION:
-                    raise RuntimeError(
-                        f'Database version higher then expected. '
-                        f'It is {db_version} should be {self._current_version}',
-                    )
+        file_version = get_file_version(latest_db_path)
 
-                # The version number in the database is smaller then the
-                # current target, this means the migration failed to execute on
-                # the last iteration, delete the partially upgraded database
-                # and start again.
-                self._delete_current_db()
```

Comment:
with a `^C`  this would not run and the old database would be lingering around. This meant we had to handle this twice, at the beginning of the method and at the end. The PR changes to only handle things at the beginning, done by `delete_dbs_with_failed_migrations`